torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1344,6 +1344,44 @@ def cast_symbool_to_symint_guardless(symbool: torch.SymBool) -> torch.SymInt:
     int_sym = sympy.Piecewise((1, symbool.node.expr), (0, True))
     return symbool.node.shape_env.create_symintnode(int_sym, hint=int(symbool.node.require_hint()))
 
+def sym_ite(pred, then_val, else_val):
+    if isinstance(pred, SymBool):
+        if pred.node.is_constant():
+            constant_pred = guard_bool(pred)
+            return then_val if constant_pred else else_val
+        return _sym_ite(pred, then_val, else_val)
```

Comment:
Check how the other sym ops go, they have to go through a dynamic dispatch on SymNode first. That allows for C++ to override the meaning of operations like this. As implemented today, this will only work if you get a Python SymNode.

Reply:
This also implies that you'll have to add a new virtual method to SymNodeImpl in C++

Reply:
Is sym_max a good example to follow? How can I test the c++ implementation?